### DISCUSSION TITLE
TypeError: __init__() takes 1 positional argument but 2 were given

### DISCUSSION THREAD

**Discussion by Emmy-Obasi** [online-ml/river#982] [Q&A] [answered]
Hello everyone,
I am trying to build a model using Gaussian Naive Bayes with these codes:

**Code**
`from river.naive_bayes import GaussianNB
from river import preprocessing
from river import feature_extraction
from river import stream
from river import metrics
from river import optim
from river import evaluate

optimizer =optim.SGD(.1)
model = (preprocessing.StandardScaler() | GaussianNB(optimizer))
metric = metrics.Accuracy()
evaluate.progressive_val_score(data, model, metric)`

but I ran into the following problem:

**Output:**
TypeError: __init__() takes 1 positional argument but 2 were given
Traceback:

File "c:\programdata\anaconda3\lib\site-packages\streamlit\script_runner.py", line 430, in _run_script
    exec(code, module.__dict__)
File "C:\codes\main.py", line 64, in <module>
    model = (preprocessing.StandardScaler() | GaussianNB(optimizer))

Can anyone help me out with this, please?

**Comment by MaxHalford** [comment, marked as answer]
Hey hey. GaussianNB doesn't have any input parameter. Here you're passing an optimizer, which doesn't work. I'm curious, where did you get the idea an optimizer was necessary?

**Reply by Emmy-Obasi** [reply]
Thanks for the prompt response.  
I am very new to machine learning and incremental ML.   However, I want to compare the performance of various classifiers when used with some optimizers.  I found this link:  https://riverml.xyz/dev/api/linear-model/LogisticRegression/
The example on that page did not also produce any result.  
Again, the documentation on Gaussian Naive Bayes for incremental learning on this page: https://riverml.xyz/dev/api/naive-bayes/GaussianNB/ is very scanty,
Kindly help me with resources that will expose me to sufficient information about GausianNB for incremental learning.  Thank you.

**Reply by MaxHalford** [reply]
> The example on that page did not also produce any result

What do you mean? What happened? Do you have a traceback? The example should work fine.

> Again, the documentation on Gaussian Naive Bayes for incremental learning on this page: https://riverml.xyz/dev/api/naive-bayes/GaussianNB/ is very scanty

I recommend looking at scikit-learn's documentation. It has more details. But our implementation is the same. In any case, you don't need an optimizer to use it, so I'm not really sure what you're trying to do here.

**Reply by Emmy-Obasi** [reply]
> **_What do you mean? What happened? Do you have a traceback? The example should work fine._**
> >I have been able to fix the bug.  I was trying to learn from the example using a different dataset
> 
> **_I recommend looking at scikit-learn's documentation. It has more details. But our implementation is the same. In any case, you don't need an optimizer to use it, so I'm not really sure what you're trying to do here._**
>>Thanks for the guide.  I am actually trying to monitor the performance of different classifiers by observing the effect of optimizers on the classifiers.  True! most examples I came across on GaussianNB did not use optimizers.  Is there any explanation for that?  I would really want to learn.  Thank you.

**Reply by MaxHalford** [reply]
Optimizers are necessary for models with weights, such as linear regression. GaussianNB is simply part of a different class of models, and is based on a different theory. You can't compare apples and oranges :)
